splash/browser_tab.py

```diff
@@ -85,6 +87,10 @@ def _init_webpage(self, verbosity, network_manager, splash_proxy_factory, render
         self.web_view.setPage(self.web_page)
         self.web_view.setAttribute(Qt.WA_DeleteOnClose, True)
 
+        self.set_window_size(render_options.get_window_size())
```

Comment:
The fact `render_options` is available to BrowserTab is a bug :) BrowserTab shouldn't be tied to GET/POST parameters extracted by a web server. 

Setting window size (and viewport size) here based on `render_options` means that the window size is set in scripts if `window_size` argument is passed to the /execute endpoint regardless of a script itself. I think we should require an explicit command for that. The fact `allowed_domains`, `proxy` and `filters` work in a similar way (see http://splash.readthedocs.org/en/latest/api.html#execute endpoint) is a bug/unfinished refactoring.